### DISCUSSION TITLE
Validate placement of each tile after placing

### DISCUSSION THREAD

**Discussion by edmund23** [flbulgarelli/headbreaker#35] [Q&A] [answered]
Hi.

First of all, nice framework, a bit to much for me, don't understand the code or better how it all combines - modules confuse me ;).

I was trying to get the tiles to validate after each connection. No luck so far. 
I tried with the targetPosition, metadata, ids and everything you get from in DOM, but can't get quite it to work.

I see there is a PieceValidator, but have no idea how and where to use it.
Ideally there would be an event onConnect and inside you could call the PieceValidator and then do something with the tile if it's not in the correct place (move it away).

Could you point me in the right direction? 
Thank you.

**Comment by flbulgarelli** [comment, marked as answer]
Hi!

Have you checked the validated puzzle sample? You can find it here: https://github.com/flbulgarelli/headbreaker/blob/93b04732021d1def33d539038f63092028fe85f9/docs/js/demo.js#L617-L645

As you mentioned, headbreaker comes with validators that allows to validate your puzzle when it changes. Both canvas and puzzle objects expose methods for registering them - `attachValidator`. Once attached, you can either check for validity manually using the `valid` property, or register one or more listeners through the `onValid` callback.
